### PR TITLE
sec: Make GraphQL cost limits configurable

### PR DESCRIPTION
**Pull request by evict**
This makes it easier for customer site-admin to change the GraphQL limits. It adds a `RateLimits` function to fetch the structure and makes sure we have default values.

## Test plan
CI tests, test locally.

### CODE REVIEW

File: schema/site.schema.json [hunk 1]
```diff
@@ -49,6 +49,27 @@
       "type": "boolean",
       "default": false
     },
+    "RateLimits": {
```
Comment (eseliger) [inline schema/site.schema.json:52]: properties in the schema are usually lowercase

```suggestion
    "rateLimits": {
```

File: cmd/frontend/graphqlbackend/graphqlbackend.go [hunk 2]
```diff
@@ -633,7 +633,7 @@ func NewSchema(
 	opts := []graphql.SchemaOpt{
 		graphql.Tracer(newRequestTracer(log.Scoped("GraphQL"), db)),
 		graphql.UseStringDescriptions(),
-		graphql.MaxDepth(maxDepth),
+		graphql.MaxDepth(conf.RateLimits().GraphQLMaxDepth),
```
Comment (eseliger) [inline cmd/frontend/graphqlbackend/graphqlbackend.go:636]: Hmm damn, this'll require a restart to take effect.. That makes rollouts much harder for multiple on-prem customers unfortunately.

I looked at the library and unfortunately the validator code lives in `internal` so we can't easily take it out of there.
Maybe a library change could help work around that, to add MaxDepthFunc that allows to return a new value for each validation pass (request).

File: internal/conf/computed.go [hunk 3]
```diff
@@ -460,6 +460,32 @@ func PasswordPolicyEnabled() bool {
...
+		}
+		return *rl
+	}
+
+	return schema.RateLimits{
+		GraphQLMaxAliases:    maxAliases,
+		GraphQLMaxFieldCount: maxFieldCount,
+		GraphQLMaxDepth:      maxDepth,
+	}
+}
```
[the remark this answers is not in the document]
Comment (evict) [reply internal/conf/computed.go:487]: TIL, thank you!

File: cmd/frontend/graphqlbackend/graphqlbackend.go (hunk 2 above)
Comment (evict) [reply cmd/frontend/graphqlbackend/graphqlbackend.go:636]: I don't think we can make that on time for the release, what do you think?

Comment (eseliger) [reply cmd/frontend/graphqlbackend/graphqlbackend.go:636]: that's okay but can we make sure to follow up on that?

**Review by eseliger — APPROVED** [review]

File: cmd/frontend/graphqlbackend/graphqlbackend.go (hunk 2 above)
Comment (evict) [reply cmd/frontend/graphqlbackend/graphqlbackend.go:636]: Will do !

**Review by mohammadualam — APPROVED** [review]

**Pull request merged**